tsl/src/bgw_policy/job.c

```diff
@@ -159,21 +178,67 @@ execute_drop_chunks_policy(int32 job_id)
 
 	/* Get the arguments from the drop_chunks_policy table */
 	args = ts_bgw_policy_drop_chunks_find_by_job(job_id);
-
+	table_relid = ts_hypertable_id_to_relid(args->fd.hypertable_id);
 	if (args == NULL)
 		ereport(ERROR,
 				(errcode(ERRCODE_TS_INTERNAL_ERROR),
 				 errmsg("could not run drop_chunks policy #%d because no args in policy table",
 						job_id)));
-
-	ts_chunk_do_drop_chunks(ts_hypertable_id_to_relid(args->fd.hypertable_id),
-							IntervalPGetDatum(&args->fd.older_than),
-							0,
-							INTERVALOID,
-							InvalidOid,
-							args->fd.cascade,
-							args->fd.cascade_to_materializations,
-							LOG);
+	// TODO for the next stage
+	if (args->fd.interval_support)
+	{
+		ts_chunk_do_drop_chunks(table_relid,
+								IntervalPGetDatum(&args->fd.older_than_interval),
+								(Datum) 0,
+								INTERVALOID,
+								InvalidOid,
+								args->fd.cascade,
+								args->fd.cascade_to_materializations,
+								LOG);
+	}
+	else
+	{
+		hcache = ts_hypertable_cache_pin();
+		hypertable = ts_hypertable_cache_get_entry(hcache, table_relid);
+		/* First verify that the hypertable corresponds to a valid table */
+		if (hypertable == NULL)
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_HYPERTABLE_NOT_EXIST),
+					 errmsg("could not set integer_now function because \"%s\" is not a hypertable",
+							get_rel_name(table_relid))));
+
+		open_dim = hyperspace_get_open_dimension(hypertable->space, 0);
+
+		Assert(IS_INTEGER_TYPE(ts_dimension_get_partition_type(open_dim)));
+
+		// todo:: perhaps use FuncnameGetCandidates instead?
+		// Q:: this uniquely identifies a function right?
+		now_func = lookup_proc_filtered(NameStr(open_dim->fd.integer_now_func_schema),
```

Comment:
This uniquely identifies a function right?
Should I use `FuncnameGetCandidates` instead of exposing `lookup_proc_filtered` ?

`lookup_proc_filtered` is pretty generic and I think it is worth exposing it.